### PR TITLE
feat(io-sign): [SFEQS-1128] Add getQtspClausesMetadata endpoint

### PR DESCRIPTION
**Pull request by grausof**
This PR adds the `getQtspClausesMetadata` endpoint of `io-sign` to #940 

#### List of Changes
Added new endpoint

#### How Has This Been Tested?
Tested locally using docker images

#### Types of changes
- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:
- [ ] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.

### CODE REVIEW

File: api_io_sign.yaml [hunk 1]
```diff
@@ -9,6 +9,22 @@ schemes:
 security:
   - Bearer: []
 paths:
+  "/qtsp/clauses":
+    post:
+      operationId: getQtspCluasesMetadata
```
Comment (gunzip) [inline api_io_sign.yaml:14]:
```suggestion
      operationId: getQtspClausesMetadata
```

File: src/services/ioSignService.ts [hunk 2]
```diff
@@ -117,4 +119,26 @@ export default class IoSignService {
...
+  > =>
+    withCatchAsInternalError(async () => {
+      const validated = await this.ioSignApiClient.getQtspClausesMetadata({});
+      return withValidatedOrInternalError(validated, response => {
+        switch (response.status) {
+          case 200:
+            return ResponseSuccessJson(response.value);
+          case 500:
+            return ResponseErrorInternal(
+              readableProblem((response.value as unknown) as ProblemJson)
```
Comment (gunzip) [inline src/services/ioSignService.ts:137]: can we decode this into `ProblemJson` instead of casting (with a` getOrElse()` fallback )?

Comment (grausof) [reply src/services/ioSignService.ts]: I added a decode with fallback. Thanks!

Comment (gunzip) [reply src/services/ioSignService.ts]: remember to add the ToDO

**Comment by gunzip** [comment]
I mean the TODO to refactor this part when the code generator is fixed

File: src/services/ioSignService.ts [hunk 3]
```diff
@@ -105,12 +108,46 @@ export default class IoSignService {
...
           case 404:
             return ResponseErrorNotFound(
-              "Resources not found",
+              resourcesNotFound,
               "The user associated with this profile could not be found."
             );
           case 500:
             return ResponseErrorInternal(
-              readableProblem((response.value as unknown) as ProblemJson)
+              // TODO: The response generated in case of Error is undefined and not a ProblemJson. [SFEQS-1199]
```
Comment (gunzip) [inline src/services/ioSignService.ts:116]:
```suggestion
              // TODO [SFEQS-1199]: When the code for openapi-codegen-ts is fixed, refactor this section. 
              // Now, it generates incorrect output whenever the http status is 500. 
```

File: src/services/ioSignService.ts [hunk 4]
```diff
@@ -105,12 +108,46 @@ export default class IoSignService {
...
+  > =>
+    withCatchAsInternalError(async () => {
+      const validated = await this.ioSignApiClient.getQtspClausesMetadata({});
+      return withValidatedOrInternalError(validated, response => {
+        switch (response.status) {
+          case 200:
+            return ResponseSuccessJson(response.value);
+          case 500:
+            return ResponseErrorInternal(
+              // TODO: The response generated in case of Error is undefined and not a ProblemJson. [SFEQS-1199]
```
Comment (gunzip) [inline src/services/ioSignService.ts:144]:
```suggestion
              // TODO [SFEQS-1199]: When the code for openapi-codegen-ts is fixed, refactor this section. 
              // Now, it generates incorrect output whenever the http status is 500. [SFEQS-1199]
```

**Review by gunzip — APPROVED** [review]

File: api_io_sign.yaml [hunk 5]
```diff
@@ -9,6 +9,22 @@ schemes:
 security:
   - Bearer: []
 paths:
+  "/qtsp/clauses":
+    post:
```
[the remark this answers is not in the document]
Comment (grausof) [reply api_io_sign.yaml:13]: Typo, sorry 😬

**Review by AleDore — APPROVED** [review]
lgtm

**Review by gquadrati — APPROVED** [review]
Just small comments

File: src/services/ioSignService.ts [hunk 6]
```diff
@@ -106,12 +110,48 @@ export default class IoSignService {
...
+    withCatchAsInternalError(async () => {
+      const validated = await this.ioSignApiClient.getQtspClausesMetadata({});
+      return withValidatedOrInternalError(validated, response => {
+        switch (response.status) {
+          case 200:
+            return ResponseSuccessJson(response.value);
+          case 500:
+            return ResponseErrorInternal(
+              // TODO [SFEQS-1199]: When the code for openapi-codegen-ts is fixed, refactor this section.
+              // Now, it generates incorrect output whenever the http status is 500. [SFEQS-1199]
```
[the remark this answers is not in the document]
Comment (grausof) [reply src/services/ioSignService.ts:164]: Types generated by `io-backend` (as written in our openapi) do not use the `ProblemJson` type so a decode is needed.

File: api_io_sign.yaml [hunk 7]
```diff
@@ -70,6 +90,42 @@ definitions:
...
+      privacy_text:
+        type: string
+        minLength: 1
+      terms_and_conditions_url:
+        type: string
+        format: url
+      nonce:
+        type: string
+        format: base64
+        minLength: 1
```
[the remark this answers is not in the document]
Comment (grausof) [reply api_io_sign.yaml:131]: you are right, i forgot the required. Added now!

**Pull request merged**